File: 2155-find-missing-observations/find-missing-observations.py
class Solution:
    def missingRolls(self, rolls: List[int], mean: int, n: int) -> List[int]:
        a=sum(rolls)
        k=len(rolls)+n
        x=(k*mean)-a
        if x> (6*n) or x<0 or x<n :return []
        else:
            res=[x//n]*n
            k=x-sum(res)
            i=0
            while k!=0:
                res[i]+=1
                i+=1
                k-=1
            return res



        

File: 2155-find-missing-observations/find-missing-observations.cpp
class Solution {
public:
    vector<int> missingRolls(vector<int>& rolls, int mean, int n) {
        long long s= accumulate(rolls.begin(),rolls.end(),0);
        int k= rolls.size()+n;
        int x= (k*mean)-s;
        vector<int>v;
        if (x<0 || x<n || x>(6*n)){
            return v;
        }
        else{
            int a=x/n;
            vector<int>res(n,a);
            int rem=x-(a*n);
            int i=0;
            while (rem!=0){
                res[i]++;
                rem--;
                i++;
            }
            return res;
        }
    }
};